libs/langchain/langchain/retrievers/you.py

```diff
@@ -0,0 +1,49 @@
+import os
+from typing import Any, Dict, List
+
+from langchain.callbacks.manager import CallbackManagerForRetrieverRun
+from langchain.pydantic_v1 import root_validator
+from langchain.schema import BaseRetriever, Document
+from langchain.utils import get_from_dict_or_env
+
+
+class YouRetriever(BaseRetriever):
+    """`You` retriever that uses You.com's search API.
+
+    To connect to the You.com api requires an API key which
+    you can get by emailing api@you.com.
+    You can check out our docs at https://documentation.you.com.
+
+    You need to set the environment variable `YDC_API_KEY` for retriever to operate.
+    """
+
+    ydc_api_key: str
+
+    @root_validator(pre=True)
+    def validate_client(
+        cls,
+        values: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        if not os.getenv("YDC_API_KEY"):
```

Comment:
i dont think we need these two lines (since users can pass it in, it doesnt need to be set, and we can rely on get_from_dict_or_env to check

Reply:
gone